frontend/src/shared/util/auth-utils.ts

```diff
@@ -0,0 +1,33 @@
+/**
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import { useAuth } from 'react-oidc-context';
+
+/**
+ * React hook to get username of current signed in user.
+ * 
+ * @returns {string}
+ */
+export const useUsername = (): string => {
```

Comment:
Why are we adding this? How is this relevant to the current issue?

This is already a large PR. If this isn't related I would apply this separately